priv/rest.ex.eex

```diff
@@ -96,43 +95,10 @@ defmodule <%= context.module_name %> do
 
     payload = encode_payload(input)
     headers = AWS.Request.sign_v4(client, method, url, headers, payload)
-    perform_request(method, url, payload, headers, options, success_status_code)
+    AWS.HTTP.request(method, url, payload, headers, options, success_status_code)
   end
 
-  defp perform_request(method, url, payload, headers, options, nil) do
-    case HTTPoison.request(method, url, payload, headers, options) do
-      {:ok, %HTTPoison.Response{status_code: 200, body: ""} = response} ->
```

Comment:
I guess we'll have to either pass the decoder as some kind of option (eg: `options[:decoder] = {AWS.XML, :decode!}`) to the AWS.HTTP module, pass an anonymous function with all this deleted code, or let it as it  is.

Reply:
@jfacorro What do you think would be the best option? I am inclined to go with 3. We get the power to replace HTTP client, but leave the validation logic for the generator.

Reply:
I moved it back, but with changes to support different HTTP clients than HTTPoison

Reply:
Sorry I didn't reply to this. I didn't notice the mention here. Great work! 👍